crates/kubelet/src/state.rs

```diff
@@ -231,42 +231,84 @@ pub trait State<PodState>: Sync + Send + 'static + std::fmt::Debug {
     ) -> anyhow::Result<serde_json::Value>;
 }
 
+async fn patch_status(api: &Api<KubePod>, name: &str, patch: serde_json::Value) {
+    match serde_json::to_vec(&patch) {
+        Ok(data) => match api.patch_status(&name, &PatchParams::default(), data).await {
+            Ok(_) => (),
+            Err(e) => {
+                warn!("Pod {} error patching status: {:?}", name, e);
+            }
+        },
+        Err(e) => {
+            warn!(
+                "Pod {} error serializing status patch {:?}: {:?}",
+                name, &patch, e
+            );
+        }
+    }
+}
+
 /// Iteratively evaluate state machine until it returns Complete.
 pub async fn run_to_completion<PodState: Send + Sync + 'static>(
     client: &kube::Client,
     state: impl State<PodState>,
     pod_state: &mut PodState,
     pod: &Pod,
-) -> anyhow::Result<()> {
```

Comment:
I'm not sure making this infallible gives us much value. We end up having to `unwrap` and we still aren't handling a status patch retry, so it feels like we just moved the problems elsewhere

Reply:
However, I did just remember that we are making state handlers infallible in #398, so I think I'm ok with merging this once we merge #398 and rebase this on top

Reply:
Yes, it's not that I was trying to make this infallible, it's that it will be once that is merged. Also, I felt that consolidating the patch to the one function in this module would make it trivial to add retry. We could go ahead and add that in this pull request if you want.

Reply:
Nope, no problem. Let's land #398 and then rebase and I'll come back and LGTM it